pkg/trace/api/otlp.go

```diff
@@ -187,6 +188,147 @@ func (o *OTLPReceiver) SetOTelAttributeTranslator(attrstrans *attributes.Transla
 
 // ReceiveResourceSpans processes the given rspans and returns the source that it identified from processing them.
 func (o *OTLPReceiver) ReceiveResourceSpans(ctx context.Context, rspans ptrace.ResourceSpans, httpHeader http.Header) source.Source {
+	if o.conf.HasFeature("enable_receive_resource_spans_v2") {
+		return o.receiveResourceSpansV2(ctx, rspans, httpHeader)
+	} else {
+		return o.receiveResourceSpansV1(ctx, rspans, httpHeader)
+	}
+}
+
+func (o *OTLPReceiver) receiveResourceSpansV2(ctx context.Context, rspans ptrace.ResourceSpans, httpHeader http.Header) source.Source {
```

Comment:
receiveResourceSpansV2 shouldn't use httpheader. Instead, we should params directly. There is only one use case in DD exporter.